h2/src/main/org/h2/index/LinkedIndex.java

```diff
@@ -127,6 +127,9 @@ public Cursor find(SessionLocal session, SearchRow first, SearchRow last) {
         String sql = builder.toString();
         try {
             PreparedStatement prep = link.execute(sql, params, false, session);
+            if (link.getFetchSize() != -1) {
+                prep.setFetchSize(link.getFetchSize());
+            }
```

Comment:
That's too late to call this method. It should be called before `PreparedStatement.executeSomething()`.

Reply:
So I must add it on the execute method in TableLInk

Reply:
Yes, that method can call it properly.

Reply:
move to execute in tablelink